File: problems-adapt/0100-0200/0106_the-missing-value/solution_xor.ts
function missingValue(nums: number[]): number {
    // Seed with n — the one index the loop below never visits — then fold
    // every index 0..n-1 and every element into one accumulator.
    let result = nums.length;
    for (let i = 0; i < nums.length; i++) {
        // Each present value matches an index and cancels it; the absent
        // value pairs with nothing and survives the fold.
        result ^= i ^ nums[i];
    }
    return result;
}

File: problems-bettercode/0201-0300/0268_missing-number/solution_xor.go
func missingNumber(nums []int) int {
	// Seed with n — the one index the loop below never visits — then fold
	// every index 0..n-1 and every element into one accumulator.
	result := len(nums)
	for i, value := range nums {
		// Each present value matches an index and cancels it; the absent
		// value pairs with nothing and survives the fold.
		result ^= i ^ value
	}
	return result
}
